src/runtime/policy/tests/policy-test.ts

```diff
@@ -490,12 +492,19 @@ policy MyPolicy {
           name,
           otherAddresses {country}
         }
+      }
+
+      policy PolicyFour {
+        from Address access {
+          state
+        }
       }`)).policies;
-    const maxReadSchema = IngressValidation.getMaxReadSchema('Person', [policy0, policy1, policy2]);
+    const maxReadSchemas = IngressValidation.getMaxReadSchemas(policies);
     const expectedSchemas = (await Manifest.parse(`
       schema Address
         number: Number
         street: Text
+        state: Text
```

Comment:
Is this really the behavior that we want?
So in Person, the policies allow `number, street, city`, 
but for Address on its own, we only allow reading `state`.
why does schema Address gets its fields permissions from referencing schemas?
my intuition is that when reading `Address` you'd only be allowed to read `state`... no?

Reply:
How is the reference to the `Address` created in both the ingresses? I am assuming that there is another particle that ingresses `Address {...}` to an `address_store` store and that the ingress particles you mentioned create references to the `Address` entries in `address_store`. In this case, don't we want to store all the fields of `Address` in `address_store`?

I agree that what fields are allowed to be read will be different depending upon where you access it from, but DFA should take care of that.

Reply:
no, the way it works today, the WritePerson particle and the WriteBusiness particle both need to have a `addresses: writes Address {...}` handle connection.
according to the ingress restriction / validation, these connections would resolve to an `address-store` with different types.

Reply:
With references it's super tricky, I agree.

All things considered I think what Gogul is doing here is the right thing to do.

With the `Address` example you bring @mmandlis it seems wrong because we have this intuition that the set of address of people does not intersect with the set of addresses of businesses (even that is a questionable assumption). However if addresses were shared (which they can be), and there was one shared particle `AddressIngestion` that is writing addresses for both people and businesses, there is nothing we can do but to write all allowed fields, as we cannot at that point know which address will references as a person's address, and which as a business address in the future.